library/src/test/kotlin/io/github/wax911/library/annotation/processor/fragment/RegexFragmentAnalyzerSpec.kt

```diff
@@ -0,0 +1,117 @@
+package io.github.wax911.library.annotation.processor.fragment
+
+import org.junit.Assert.assertEquals
+import org.junit.Test
+
+class RegexFragmentAnalyzerSpec {
+    private val fragmentNameTemplate = "someObject%sFragment"
+    private val typeTemplate = "Object%s"
+    private val fragmentA = createFragmentName("A")
+    private val fragmentB = createFragmentName("B")
+    private val fragmentC = createFragmentName("C")
+    private val fragmentD = createFragmentName("D")
+    private val typeA = createType("A")
+    private val typeB = createType("B")
+    private val typeC = createType("C")
+    private val typeD = createType("D")
+
+    private val fragmentTemplate = """
+        fragment %s on %s {
+          id
+          name
+          %s
+        }
+    """.trimIndent()
+
+    private val queryTemplate = """
+        query SomeQuery {
+          someQuery {
+            objectA {
+              ...$fragmentA
+            }
+            objectB {
+              ...$fragmentB
+            }
+            objectC {
+              ...$fragmentC
+            }
+          }
+        }
+
+        %s
+    """.trimIndent()
+
+    private val subj = RegexFragmentAnalyzer()
+
+    @Test
+    fun `Given a query with all fragments defined, When analyze fragments, Then analysis matches expectation`() {
+        val expected = setOf(
+            FragmentAnalysis(fragmentA, true),
+            FragmentAnalysis(fragmentB, true),
+            FragmentAnalysis(fragmentC, true)
+        )
+
+        val definedFragments = """
+            ${createFragment(fragmentA, typeA)}
+
+            ${createFragment(fragmentB, typeB)}
+
+            ${createFragment(fragmentC, typeC)}
+        """.trimIndent()
+
+        assertEquals(expected, subj.analyzeFragments(createQuery(definedFragments)))
+    }
+
+    @Test
+    fun `Given a query with 1 of 3 fragments defined, When analyze fragments, Then analysis matches expectation`() {
+        val expected = setOf(
+            FragmentAnalysis(fragmentA, true),
+            FragmentAnalysis(fragmentB, false),
+            FragmentAnalysis(fragmentC, false),
+            FragmentAnalysis(fragmentD, true)
+        )
+
+        val definedFragments = """
+            ${createFragment(fragmentA, typeA, "...$fragmentD")}
+            ${createFragment(fragmentD, typeD)}
+        """.trimIndent()
+
+        println(definedFragments)
+
+        assertEquals(expected, subj.analyzeFragments(createQuery(definedFragments)))
+    }
+
+    @Test
+    fun `Given a query with no fragments defined, When analyze fragments, Then analysis matches expectation`() {
+        val expected = setOf(
+            FragmentAnalysis(fragmentA, false),
+            FragmentAnalysis(fragmentB, false),
+            FragmentAnalysis(fragmentC, false)
+        )
+
+        assertEquals(expected, subj.analyzeFragments(createQuery()))
+    }
+
+    @Test
```

Comment:
this test is a copy of the one above, but the Given statements are different?

Reply:
Good looking out! Yeah, that's totally wrong. I think I broke/changed it when refactoring this test to clean it up a bit.

I'll make sure to fix this in the next PR.

Reply:
https://github.com/Ibotta/retrofit-graphql/commit/4f82d76993fc2897f244a09e022de153cf53de35

(First commit of the soon-to-be next PR)